Modules/getpath.py

```diff
@@ -509,11 +510,12 @@ def search_up(prefix, *landmarks, test=isfile):
             build_stdlib_prefix = build_prefix
         else:
             build_stdlib_prefix = search_up(build_prefix, *BUILDSTDLIB_LANDMARKS)
-        # Always use the build prefix for stdlib
-        if build_stdlib_prefix:
-            stdlib_dir = joinpath(build_stdlib_prefix, 'Lib')
-        else:
-            stdlib_dir = joinpath(build_prefix, 'Lib')
+        # Use the build prefix for stdlib when not explicitly set
+        if not stdlib_dir_was_set_in_config:
```

Comment:
`stdlib_dir` isn't set by this point unless it was in the config, so we can just check that it's empty and set it if so.

Later on there are more checks that may overwrite `stdlib_dir` without checking. Those also ought to check that it's empty.

`stdlib_dir_was_set_in_config` shouldn't be needed, I don't think. But it might be (if e.g. we need one of the later `stdlib_dir` assignments to overwrite an earlier one if it were set from a build directory, but that shouldn't be the case).

Reply:
Thanks for the review!

Before making a change, I added `stdlib_dir_was_set_in_config` for this case: https://github.com/python/cpython/blob/a0773b89dfe5cd2190d539905dd89e7f6455668e/Modules/getpath.py#L549 as it was unconditionally setting `stdlib_dir` to `None`. At this point, `stdlib_dir` might be already set either in the config or some conditions above. So `stdlib_dir_was_set_in_config` is needed I think.

And given `stdlib_dir_was_set_in_config` is needed here, do you prefer using `stdlib_dir_was_set_in_config` everywhere, or check if `stdlib_dir` is empty in some cases?

Reply:
You'll need to determine whether setting `home` _should_ override other parts of the config. Historically, I believe that's what it has done, and so arguably it should continue to do it. Or maybe it should change? But that will need to be a deliberate (and documented) change of behaviour.

Any changes made to this script are _incredibly_ fragile. Hopefully if this change goes in then we'll have all of the 3.13 prereleases to find out who it broke, but it would be best to figure out which change is least breaking, and be very explicit about which scenarios will change. Even then, we're basically guaranteed to miss some things.

Reply:
Thanks, I reworded the changelog to say "When PyConfig.stdlib_dir is explicitly set, it's now respected and won't be overridden by PyConfig.home."

For `stdlib_dir_was_set_in_config`, I _can_ test if stdlib_dir is None _here_. But below when `home` is set, it will override the stdlib_dir found by the build directory here. And I don't want to change this behavior so `stdlib_dir_was_set_in_config` is necessary there (similarly the two places below). Since there is `stdlib_dir_was_set_in_config`, I'm using it everywhere. Let me know if you prefer testing test if stdlib_dir is None _here_.